backend/app/api/routes/patients.py

```
@@ -0,0 +1,71 @@
+import uuid
+
+
+from app import crud
+from fastapi import APIRouter, HTTPException
+from sqlmodel import func, select
+
+from app.api.deps import CurrentUser, SessionDep
+from app.models import Patient, PatientCreate, PatientPublic, PatientsPublic, PatientUpdate, Message
+
+router = APIRouter(prefix="/patients", tags=["patients"])
+
+
+@router.get("/", response_model=PatientsPublic)
+def read_patients(
+    session: SessionDep, current_user: CurrentUser, 
+) -> PatientsPublic:
+   """
+get patients of currentuser
+    """
+   patients =  crud.get_caregiver_patients(session=session,caregiver=current_user)
+   return PatientsPublic(data=patients, count = len(patients))
+
+
+
+@router.get("/{id}", response_model=PatientPublic)
+def read_patient(session: SessionDep, current_user: CurrentUser,id:uuid.UUID) -> PatientPublic | None:
+    """
+    Get patient.
+    """
+    patient  = crud.get_patient(session=session,id=id)
+    if current_user.id == patient.owner_id:
+        return PatientPublic.form_patient(patient=patient)
+    else:
+        return None
```

please raise an HTTP exception

---

✅